[AOTInductor] Small refactor so both Meta internal and OSS can deal with misplaced args and kwargs for Extern Fallback kernels

hl475 opened the pull request:
Summary:
In torch/_inductor/lowering.py (https://fburl.com/code/jd58vxpw), we are using
```
fallback_cumsum(x, dim=axis, dtype=dtype)
```
so this will treat `x` as args, `dim` and `dtype` as kwargs from https://fburl.com/code/cikchxp9


The issue has been fixed from D52530506 for OSS but not Meta internal. This diff address the Meta internal issue by some refactoring so both Meta internal and OSS can use the same helper function. The diff also added some debug log.

Test Plan:
before
```
aoti_torch_proxy_executor_call_function(proxy_executor, 2, 1, std::vector<int64_t>{torch.int64}.data(), 2, std::vector<AtenTensorHandle>{buf702, buf708}.data());
```
after
```
aoti_torch_proxy_executor_call_function(proxy_executor, 2, 1, std::vector<int64_t>{0}.data(), 2, std::vector<AtenTensorHandle>{buf702, buf708}.data());
```
so `torch.int64` changed to `0`

Differential Revision: D52532031




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

hl475 commented:
@pytorchbot label "topic: not user facing"

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4485,17 +4485,32 @@ def is_not_write(arg):
         pos_arg_name = self.args_default_value[pos]["name"]
         if pos_arg_name in kwargs:
+            log.debug(
+                f"Found argument {pos_arg_name} with value {kwargs[pos_arg_name]} from kwargs"  # noqa: G004
```
nit - I think we discourage using f-strings in log statements.

hl475 replied:
Thanks! I changed this in the next exported diff :)

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4504,7 +4519,11 @@ def get_pos_arg_value(self, pos, kwargs):
-        return self.args_default_value[pos]["value"]
+        arg_default_value = self.args_default_value[pos]["value"]
+        log.debug(
+            f"Use default value {arg_default_value} for argument {pos_arg_name}"  # noqa: G004
```
nit - no f-str in log statements

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4612,6 +4622,28 @@ def get_alias_names(self):
+        # if they're in the kwargs or use their default value
+        if n_args < n_pos_args:
+            log.debug(
+                f"{self.op_overload} has {n_pos_args-n_args} unprovided positional arguments. "  # noqa: G004
```
nit - no f-str

hl475 committed:
[AOTInductor] Small refactor so both Meta internal and OSS can deal with misplaced args and kwargs for Extern Fallback kernels (#116779)

Summary:

In torch/_inductor/lowering.py (https://fburl.com/code/jd58vxpw), we are using
```
fallback_cumsum(x, dim=axis, dtype=dtype)
```
so this will treat `x` as args, `dim` and `dtype` as kwargs from https://fburl.com/code/cikchxp9


The issue has been fixed from D52530506 for OSS but not Meta internal. This diff address the Meta internal issue by some refactoring so both Meta internal and OSS can use the same helper function. The diff also added some debug log.

Test Plan:
before
```
aoti_torch_proxy_executor_call_function(proxy_executor, 2, 1, std::vector<int64_t>{torch.int64}.data(), 2, std::vector<AtenTensorHandle>{buf702, buf708}.data());
```
after
```
aoti_torch_proxy_executor_call_function(proxy_executor, 2, 1, std::vector<int64_t>{0}.data(), 2, std::vector<AtenTensorHandle>{buf702, buf708}.data());
```
so `torch.int64` changed to `0`

Reviewed By: desertfire

Differential Revision: D52532031

hl475 commented:
@pytorchbot merge
